File: Group6System-main/Group6System-main/ApplicationTier/src/main/java/com/example/applicationtier/models/Transaction.java
package com.example.applicationtier.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.springframework.stereotype.Component;

import java.util.Date;

@Component
public class Transaction {
    @JsonProperty("sender")
    private Account sender;
    @JsonProperty("receiver")
    private Account receiver;
    @JsonProperty("amount")
    private double amount;
    @JsonProperty("message")
    private String message;
    @JsonProperty("date")
    private String date;
    @JsonProperty("save")
    private boolean save;

    public Transaction() {
    }

    @Override
    public String toString() {
        return "Transaction{" +
                "sender=" + sender +
                ", receiver=" + receiver +
                ", amount=" + amount +
                ", message='" + message + '\'' +
                ", date=" + date +
                ", save=" + save +
                '}';
    }

    public Account getSender() {
        return sender;
    }

    public void setSender(Account sender) {
        this.sender = sender;
    }

    public Account getReceiver() {
        return receiver;
    }

    public void setReceiver(Account receiver) {
        this.receiver = receiver;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public boolean isSave() {
        return save;
    }

    public void setSave(boolean save) {
        this.save = save;
    }

}

File: Group6System-main/Group6System-main/BlazorClient/BlazorClient/Models/Transaction.cs
using System;
using System.Text.Json.Serialization;

namespace BlazorClient.Models
{
    public class Transaction
    {
        [JsonPropertyName("transactionnumber")]
        public int TransactionNumber { get; set; }
        
        [JsonPropertyName("sender")]
        public Account Sender { get; set; }
        
        [JsonPropertyName("receiver")]
        public Account Receiver { get; set; }
        
        [JsonPropertyName("amount")]
        public double Amount { get; set; }
        
        [JsonPropertyName("message")]
        public String Message { get; set; }
        
        [JsonPropertyName("date")]
        public string Date { get; set; }
        
        [JsonPropertyName("save")]
        public bool Save { get; set; }

        public Transaction(Account sender, Account receiver, double amount, string message, string date, bool save)
        {
            Sender = sender;
            Receiver = receiver;
            Amount = amount;
            Message = message;
            Date = date;
            Save = save;
        }
    }
}